File: app/src/main/java/thames_tide_provider/web/configurationitems/MinuteStep.java
package thames_tide_provider.web.configurationitems;


public enum MinuteStep {
    ONE(1),
    FIVE(5),
    TEN(10),
    FIFTEEN(15),
    HALF_HOUR(30),
    HOUR(60);

    private final int step;

    MinuteStep(int step) {
        this.step = step;
    }

    public int getStep() {
        return step;
    }
}

File: app/src/main/java/com/bearcave/passageplanning/tides/web/configurationitems/MinuteStep.kt
package com.bearcave.passageplanning.tides.web.configurationitems

import java.util.*


enum class MinuteStep constructor(val value: Int) {
    ONE(1),
    FIVE(5),
    TEN(10),
    FIFTEEN(15),
    HALF_HOUR(30),
    HOUR(60);

    companion object {
        fun getByValue(value: Int): MinuteStep {
            values()
                    .filter { it.value == value }
                    .forEach { return it }

            throw NoSuchElementException()
        }
    }
}
